---
output: 
  html_document:
---

<style type="text/css">
  body{
  font-size: 13pt;
}
</style>



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(tidyverse)
library(ggsignif)
library(emmeans)
library(viridis)
#library(brms)
library(ape)
library(phangorn)
library(castor)
#install.packages("BiocManager")
#BiocManager::install("ggtree")
library(ggtree)
library(RColorBrewer)
library(patchwork)
library(kableExtra)
library(DT)
library(plotly)
#library(rredlist)
conWLAB <- read_csv("data/conWLAB.csv")
prunedtree <- read.tree("data/prunedtree.tre")
HPAI_withold <- read_csv("data/HPAI_withold.csv")
```
<br />

### Australian bird species at risk of HPAI Clade 2.3.4.4b (H5 bird flu) after an incursion event

<br />

This part of AviFluMap applies a family-level phylogenetic model using the H5 HPAI events from the current panzootic, to determine the phylogenetic signal in HPAI susceptibility to infection (and subsequent sickness and death). The model is based on event data retrieved from WAHIS provided by the World Organisation for Animal Health (WOAH) as at 06/01/2025. The model uses this H5 HPAI event data to predict the likely prevalence of infections in Australian species. 

**IMPORTANT:** Please see the caveats on using this tool in *About the Data* to learn more about the data sets including disclaimers, and assumptions and limitations of the modelling output. Users should revisit the site regularly to access the latest analysis based on the latest data in WAHIS. When the data used for AviFluMap is updated, the analysis may show revised results (such as new species at risk). Data will continue to be updated to AviFluMap. However at present the schedule for future updates has not yet been determined.

<br />

### Defining *HPAI Susceptibility*

<br />

In the context of AviFluMap, the term “HPAI susceptibility” refers to the likelihood that a given bird species will become infected with H5 bird flu and subsequently develop disease due to infection, act as a reservoir for its maintenance, and/or assist in the spread of the virus. 
Because the majority (~98%) of H5 HPAI wild bird events in the WAHIS database since October 2021 report deaths for the species recorded, AviFluMap assumes that a species’ predicted susceptibility to infection is also closely linked to a species predicted likelihood of experiencing sickness and death (referred to as “Species at Risk”). 

It is important to note that the species risk shown by AviFluMap does not consider individual population dynamics, conservation status, or other threatening processes that may affect population vulnerability. Consequently, “the Species at Risk Index” should not be used in isolation to predict a species population level impact of H5 bird flu. 

To a lesser extent, AviFluMap may also identify species that are less likely to experience sickness and death but could still play a role in maintaining or spreading the virus as reservoirs or carriers. Note that reservoir species, species that disperse the virus and species that experience sickness or death (individually or at the species level) are not necessarily the same.

Since genetically related species also share physiological, behavioural and ecological characteristics, the approach of using family-level phylogenetic signal also serves as a proxy for species sharing similar habitats/ecologies. Thus, it is assumed that the phylogeny of birds also informs the specific environmental conditions they prefer (e.g. freshwater habitats), their specific behaviours (e.g. colony breeding), and foraging strategies (e.g. scavenging) and in turn HPAI susceptibility. In other words, it is assumed that all species in the same family are similarly susceptible to HPAI infection and subsequent sickness and death, regardless of how common those species are and how often they may have or may not have been tested for HPAI. For further explanation on the validity of this approach, see the *About the Data* for a full list of assumptions relating to this analysis.

Since species’ HPAI susceptibility is likely determined by the preferred environmental conditions, specific behaviours, foraging strategies and other behavioural and ecological conditions, it should be stressed that the analysis predicts HPAI susceptibility in *natural* settings and not necessarily for birds in captivity. It should be assumed that the HPAI virus *can* infect all species of birds.

<br />

### Step one: modelling outbreak prevalence among avian families

<br />

The first step in the analysis is to model the number of H5 HPAI events of the current avian influenza panzootic to determine the phylogenetic signal. The dataset consists of all H5 HPAI events in wild birds reported to the WAHIS database  since October 2021, all of which have occurred outside of Australia. While not all events are confirmed to be HPAI clade 2.3.4.4b, this is most likely the case for the vast majority of wild bird events. The model includes a family-level phylogeny of affected birds, from <a href="https://doi.org/10.1093/molbev/msaa191" target="_blank">Kuhl et al. 2021</a>.

<br />


```{r, fig.width=8, fig.height=8}
phyl2 <- read.nexus("data/output.nex")
phylcon2 <- maxCladeCred(phyl2)
PhylSp <- unique(phylcon2$tip.label)
keep <- as.vector(PhylSp[(PhylSp %in% HPAI_withold$scientificName)])
phyl2 <- drop.tip(phylcon2, phylcon2$tip.label[-match(keep,phylcon2$tip.label)])
PhylSp2 <- unique(phyl2$tip.label)
HPAIold_raw <- HPAI_withold %>% select(scientificName, Order, Outbreaks)
```

```{r, fig.width=10, fig.height=10}
HPAIold_raw <- HPAIold_raw %>% 
  mutate(index = match(scientificName, phylcon2$tip.label))
p <- ggtree(phylcon2, layout = "circular") %<+% HPAIold_raw + 
  geom_tippoint(aes(color = log10(Outbreaks), size = log10(Outbreaks))) + 
  scale_color_gradient(low= "blue", high = "red",
                       name = "Number of HPAI notifications (logged)") + 
  scale_size_continuous(range = c(0.5,4), breaks = c(500,100,5))+
  theme(legend.position = "right", legend.key.size = unit(0.5,"cm"),
        legend.text = element_text(size= 10), 
        legend.title = element_text(size = 12))+ 
  xlim(0,150)+ 
  guides(size=guide_legend(title="Number of HPAI notifications (logged)"))
species_to_order <- HPAIold_raw %>%
  filter(scientificName %in% phylcon2$tip.label) %>%
  distinct(scientificName, Order)
tip_order <- p$data %>%
  filter(isTip) %>%
  left_join(species_to_order, by = c("label" = "scientificName"))
order_labels <- tip_order %>%
  group_by(Order.x) %>%
  summarize(
    angle = mean(y),
    x = max(x) + 5,  # Place just outside the tree's edge
    y = mean(y)
  ) %>% 
  mutate(
    adj_angle = ifelse(angle > 90 & angle < 280, angle + 180, angle),  # Flip angles
    hjust = ifelse(angle > 90 & angle < 280, 1, 0)  # Adjust horizontal alignment
  )

p +
  geom_text(
    data = order_labels,
    aes(x = x, y = y, label = Order.x, angle = adj_angle, hjust = hjust),
    #    vjust = 2,   # Adjust vertical alignment
    size = 4
  )
 
```

The model reports a statistically significant phylogenetic signal (on a scale of 0-1, Pagel's lambda: 0.91, 95% CI: 0.54 - 1.00), and the phylogeny shows clusterings of outbreaks amongst **ducks and geese** (Anseriformes), **seabirds** (Suliformes), **shorebirds** (Charadriiformes), **birds of prey** (Accipitriformes and Falconiformes), and **crows** (within Passeriformes).

<br />

### Step two: Predicting HPAI susceptibility in Australian species

<br />

The above model was then used to predict which Australian species are likely to be particularly susceptible to HPAI infection, again using a family-level phylogeny. The list of Australian birds is the <a href="https://birdata.birdlife.org.au/whats-in-a-name" target="_blank">BirdLife Working List of Australian Birds</a>, which has been pruned to exclude rare vagrants and very uncommon non-native species. Susceptibility to HPAI infection was expressed as an “HPAI Susceptibility Index” on a scale from 0-1. This score was based on the predicted number of HPAI events by the model outlined above, where 80 predicted events represented a susceptibility index of 1. Any species that had an estimated HPAI susceptibility index of 0.5 or more was deemed as being moderately susceptible to HPAI infection and 0.8 or more as being highly susceptible to HPAI infection. A susceptibility index of 0.99 corresponds to 79 predicted HPAI events (in a family).

It should be noted that the predictions of HPAI susceptibility may change as more data becomes available. Therefore, this index should be used as an indicator of HPAI susceptible families. As with any empirical modelling, the predictions are likely more accurate for the families that are well represented in the data (such as Sulidae and Laridae) than the ones with little or no data (such as Anhingidae).

<br />


```{r, fig.width=25, fig.height=25}
fams <- unique(prunedtree$tip.label)
famsDS <- unique(HPAI_withold$Family)
famsAus <- unique(conWLAB$Family)
famsNew <- famsAus[!(famsAus %in% famsDS)]
famsBoth <- c(famsDS, famsAus)
#famsBoth[!(famsBoth %in% fams)]
todrop2 <- fams[!(fams %in% famsBoth)]
prunedtree3 <- drop.tip(prunedtree, todrop2)
```


```{r, results='hide'}
FamilyAvg <- HPAI_withold %>% 
  group_by(Family) %>% summarise(Outbreaks = mean(Outbreaks))
new <- famsAus[!(famsAus %in% famsDS)]
new_ds <- data.frame(Family = new, Outbreaks = NA)
FamilyAvgFull <- rbind(FamilyAvg, new_ds)
FamilyAvgFull <- FamilyAvgFull[match(prunedtree3$tip.label, FamilyAvgFull$Family), ]
all.equal(prunedtree3$tip.label, FamilyAvgFull$Family)
outbreaksdata <- FamilyAvgFull$Outbreaks

predictedPIC <- hsp_independent_contrasts(prunedtree3, outbreaksdata)

# Extract the tip-specific values from the results
tip_states <- predictedPIC$states[1:length(prunedtree3$tip.label)]
# Combine with tip labels to create a dataframe
tip_results <- data.frame(Family = prunedtree3$tip.label, Outbreaks = tip_states)

tip_resultsAus <- tip_results %>% filter(Family %in% famsAus)

normalize_log <- function(x) {
  log_x <- log(x + 1)  # Add 1 to avoid log(0)
  return((log_x - min(log_x)) / (4.4 - min(log_x))) #edited max to make highest scale 0.99, rather than 1.0
}

# Apply normalization
tip_resultsAus <- tip_resultsAus %>% 
  mutate(HPAISusceptibilityIndex = normalize_log(Outbreaks))
write.csv(tip_resultsAus, "tip_resultsAus.csv", row.names = FALSE)
```

```{r, fig.width=10, fig.height=10}
todrop2 <- fams[!(fams %in% famsAus)]
prunedtree4 <- drop.tip(prunedtree, todrop2)

ggtree(prunedtree4, layout = "circular", branch.length = "none") %<+% tip_resultsAus + 
  geom_tiplab(size = 3.5)+
  geom_tippoint(aes(color = HPAISusceptibilityIndex, size = HPAISusceptibilityIndex)) + 
  scale_color_gradient(low= "blue", high = "red",
                       name = "Predicted HPAI susceptibility") + 
  scale_size_continuous(range = c(0.5,4), breaks = c(0.01, 0.5, 0.99))+
  theme(legend.position = "right", legend.key.size = unit(0.5,"cm"),
        legend.text = element_text(size= 10), 
        legend.title = element_text(size = 12))+ 
  xlim(0,30)+ 
  guides(size=guide_legend(title="Predicted HPAI susceptibility"))
```

<br />

Lastly, the table below presents a full list of HPAI outbreak susceptibility for all Australian bird families. The list is currently ordered by families with the highest estimated outbreaks, but can be ordered based on other columns by interacting with the column headings. The table can also be downloaded using a range of formats.

<br />

```{r, fig.width=20, fig.height=20}
predicted <- conWLAB %>% 
  left_join(tip_resultsAus, by = "Family")
predicted <- predicted %>% arrange(desc(HPAISusceptibilityIndex))

treedat <- predicted %>% select(Order, FamilyCommonName, FamilyScientificName, HPAISusceptibilityIndex)

treedat <- treedat %>% rename("Family (common name)" = "FamilyCommonName",
                              "Family (scientific name)" = "FamilyScientificName",
                              "Estimated HPAI Susceptibility Index" = "HPAISusceptibilityIndex")
datatable(treedat, options = list(iDisplayLength = 25, dom = 'Blfrtip', buttons = c('copy', 'csv', 'excel', 'pdf')), extensions = 'Buttons') %>% 
  formatRound(columns = "Estimated HPAI Susceptibility Index", digits = 2)
```

<br />

<a href="https://vhost2009.hosted-sites.deakin.edu.au/HPAI/Species-predictions-list.html" target="_blank"> Click here</a> for a full list of Australian species belonging to these families, their key ecological traits, and their conservation status.

<br />

<br />

### Why using family-level phylogeny makes sense for predicting HPAI susceptibility

<br />

In the approach outlined above, family-level phylogeny is used to understand and predict HPAI susceptibility. Some of the feedback received to date on AviFluMap has questioned why ecological traits (such as habitat, diet, and tendency to congregate) were not used in the model. Below, is an outline of about why using ecological traits may be confounding, and why family very adequately accounts for much ecological information on the various species.

The heatmap below shows total number of HPAI events in birds with different diet and habitats, separated by whether they congregate or not. The numbers inside the tiles will update automatically when a family is chosen. When “All” is selected, it shows the total number of species that fit that description. When one of the key families is selected, it will show the proportion of species in that category that are part of the selected family. Furthermore, when a family is selected, the heatmap will update to show total number of notifications for that family.

The aim of this is to show that broad categorisations like “freshwater” or “predator” are not insightful by themselves - these interact with other categories. For example, it is only congregating freshwater birds with plant-based diets that have many HPAI notifications. By selecting different families, it should become clear how Family, as a taxonomic grouping, accounts for many such interactions - ducks disproportionately account for the freshwater birds with plant-based diets, for example. Hence, using family-level phylogeny takes into account a considerable amount of ecological information, and therefore provides a valid basis for predictions of which Australian birds may be susceptible to HPAI infection.

While using family-level phylogeny thus makes much sense for predicting HPAI susceptibility for individual species, exceptions might apply, and predictions should be considered carefully. For example, it might be expected that an Australian wood duck would be less susceptible to HPAI infection than other ducks, because it has a different ecology to other members of its taxonomic family. For this reason, decisions on wildlife risk mitigation strategies should be cautiously based on the susceptibility predictions herein, and with an understanding of uncertainty. Especially when considering species that have thus far not experienced exposure to H5 bird flu, which includes many species native to Australia. 
 

<br />


```{r, fig.width = 10, fig.height=6}

# All
H1t <- HPAI_withold %>% filter(CongregateTrue== "Y") %>% group_by(Habitat_Groupings, Diet_Groupings, Family) %>% 
  summarise(TotalNotifications = sum(Outbreaks), n = n())
H1nt <- HPAI_withold %>% filter(CongregateTrue== "N") %>% group_by(Habitat_Groupings, Diet_Groupings, Family) %>% 
  summarise(TotalNotifications = sum(Outbreaks), n = n())
AllCong <- HPAI_withold %>% filter(CongregateTrue== "Y") %>% group_by(Habitat_Groupings, Diet_Groupings) %>% 
  summarise(TotalNotifications = sum(Outbreaks), n = n())
AllNoCong <- HPAI_withold %>% filter(CongregateTrue== "N") %>% group_by(Habitat_Groupings, Diet_Groupings) %>% 
  summarise(TotalNotifications = sum(Outbreaks), n = n())

# Anatidae
AnatidaeT <- H1t %>% mutate(prop = case_when(Family == "Anatidae" ~ "Yes", TRUE ~ "No"))
AnatidaeT <- AnatidaeT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
AnatidaeT$prop2 <- format(round(AnatidaeT$prop2, 2),nsmall = 2)

AnatidaeNT <- H1nt %>% mutate(prop = case_when(Family == "Anatidae" ~ "Yes", TRUE ~ "No"))
AnatidaeNT <- AnatidaeNT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
AnatidaeNT$prop2 <- format(round(AnatidaeNT$prop2, 2),nsmall = 2)

# Laridae
LaridaeT <- H1t %>% mutate(prop = case_when(Family == "Laridae" ~ "Yes", TRUE ~ "No"))
LaridaeT <- LaridaeT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
LaridaeT$prop2 <- format(round(LaridaeT$prop2, 2),nsmall = 2)

LaridaeNT <- H1nt %>% mutate(prop = case_when(Family == "Laridae" ~ "Yes", TRUE ~ "No"))
LaridaeNT <- LaridaeNT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
LaridaeNT$prop2 <- format(round(LaridaeNT$prop2, 2),nsmall = 2)

# Sulidae
SulidaeT <- H1t %>% mutate(prop = case_when(Family == "Sulidae" ~ "Yes", TRUE ~ "No"))
SulidaeT <- SulidaeT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
SulidaeT$prop2 <- format(round(SulidaeT$prop2, 2),nsmall = 2)

SulidaeNT <- H1nt %>% mutate(prop = case_when(Family == "Sulidae" ~ "Yes", TRUE ~ "No"))
SulidaeNT <- SulidaeNT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
SulidaeNT$prop2 <- format(round(SulidaeNT$prop2, 2),nsmall = 2)

# Accipitridae
AccipitridaeT <- H1t %>% mutate(prop = case_when(Family == "Accipitridae" ~ "Yes", TRUE ~ "No"))
AccipitridaeT <- AccipitridaeT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
AccipitridaeT$prop2 <- format(round(AccipitridaeT$prop2, 2),nsmall = 2)

AccipitridaeNT <- H1nt %>% mutate(prop = case_when(Family == "Accipitridae" ~ "Yes", TRUE ~ "No"))
AccipitridaeNT <- AccipitridaeNT  %>% group_by(Habitat_Groupings, Diet_Groupings) %>%
summarise(tot2 = sum(TotalNotifications), prop2 = sum(TotalNotifications[prop == "Yes"])/((sum(TotalNotifications[prop == "Yes"])+sum(TotalNotifications[prop == "No"]))), tot3 = sum(TotalNotifications[prop == "Yes"]))
AccipitridaeNT$prop2 <- format(round(AccipitridaeNT$prop2, 2),nsmall = 2)

# Congregating birds
p <- plot_ly(
  data = AllCong,
  x = ~Habitat_Groupings,
  y = ~Diet_Groupings,
  z = ~TotalNotifications,  # Initial data (z1)
  type = "heatmap",
  zauto = FALSE,
  zmin = 0,
  zmax = 2000,
  text = ~n,  # Initial labels (text_labels1)
  hoverinfo = 'z+text',
  colors = colorRamp(c("blue","red")),
  texttemplate = "%{text}",  # Display the text in the center of each tile
  textfont = list(color = "white", size = 18),  # Customize font color and size
  showscale = TRUE
) %>%
  layout(
    title = "Congregating birds",
    xaxis = list(title = "Habitat groupings"),
    yaxis = list(title = "Diet groupings"),
    coloraxis = list(
      colorbar = list(
        title = "Total outbreaks",  # Change the legend title here
        tickvals = seq(1, 2000, 500),
        ticktext = c("500", "1000", "1500", "2000"))),
    updatemenus = list(
      list(
        type = 'buttons',
        buttons = list(
          list(
            method = "restyle",
            args = list(list(x = list(AllCong$Habitat_Groupings), y = list(AllCong$Diet_Groupings), z = list(AllCong$TotalNotifications), text = list(AllCong$n))),
            label = "All"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(AnatidaeT$Habitat_Groupings), y = list(AnatidaeT$Diet_Groupings), z = list(AnatidaeT$tot3), text = list(AnatidaeT$prop2))),
            label = "Anatidae (ducks and geese)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(LaridaeT$Habitat_Groupings), y = list(LaridaeT$Diet_Groupings), z = list(LaridaeT$tot3), text = list(LaridaeT$prop2))),
            label = "Laridae (gulls and terns)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(SulidaeT$Habitat_Groupings), y = list(SulidaeT$Diet_Groupings), z = list(SulidaeT$tot3), text = list(SulidaeT$prop2))),
            label = "Sulidae (gannets and boobies)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(AccipitridaeT$Habitat_Groupings), y = list(AccipitridaeT$Diet_Groupings), z = list(AccipitridaeT$tot3), text = list(AccipitridaeT$prop2))),
            label = "Accipitridae (eagles, kites, and goshawks)"
          )
        ),
          x = -1,
          y = 1,
          xanchor = "left",
          yanchor = "top"
      )
    )
  )

p

# Non-Congregating birds
np <- plot_ly(
  data = AllNoCong,
  x = ~Habitat_Groupings,
  y = ~Diet_Groupings,
  z = ~TotalNotifications,  # Initial data (z1)
  type = "heatmap",
  zauto = FALSE,
  zmin = 0,
  zmax = 2000,
  text = ~n,  # Initial labels (text_labels1)
  hoverinfo = 'z+text',
  colors = colorRamp(c("blue","red")),
  texttemplate = "%{text}",  # Display the text in the center of each tile
  textfont = list(color = "white", size = 18),  # Customize font color and size
  showscale = TRUE
) %>%
  layout(
    title = "Non-congregating birds",
    xaxis = list(title = "Habitat groupings"),
    yaxis = list(title = "Diet groupings"),
    coloraxis = list(
      colorbar = list(
        title = "Total outbreaks",  # Change the legend title here
        tickvals = seq(1, 2000, 500),
        ticktext = c("500", "1000", "1500", "2000"))),
    updatemenus = list(
      list(
        type = 'buttons',
        buttons = list(
          list(
            method = "restyle",
            args = list(list(x = list(AllNoCong$Habitat_Groupings), y = list(AllNoCong$Diet_Groupings), z = list(AllNoCong$TotalNotifications), text = list(AllNoCong$n))),
            label = "All"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(AnatidaeNT$Habitat_Groupings), y = list(AnatidaeNT$Diet_Groupings), z = list(AnatidaeNT$tot3), text = list(AnatidaeNT$prop2))),
            label = "Anatidae (ducks and geese)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(LaridaeNT$Habitat_Groupings), y = list(LaridaeNT$Diet_Groupings), z = list(LaridaeNT$tot3), text = list(LaridaeNT$prop2))),
            label = "Laridae (gulls and terns)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(SulidaeNT$Habitat_Groupings), y = list(SulidaeNT$Diet_Groupings), z = list(SulidaeNT$tot3), text = list(SulidaeNT$prop2))),
            label = "Sulidae (gannets and boobies)"
          ),
          list(
            method = "restyle",
            args = list(list(x = list(AccipitridaeNT$Habitat_Groupings), y = list(AccipitridaeNT$Diet_Groupings), z = list(AccipitridaeNT$tot3), text = list(AccipitridaeNT$prop2))),
            label = "Accipitridae (eagles, kites, and goshawks)"
          )
        ),
          x = -1,
          y = 1,
          xanchor = "left",
          yanchor = "top"
      )
    )
  )

np

```



